s3/libirods_s3.cpp

```diff
@@ -226,13 +228,23 @@ static char *s3CalcMD5( int fd, off_t start, off_t length )
 }
 // Increment through all specified hostnames in the list, locking in the case
 // where we may be multithreaded
-static const char *s3GetHostname()
+static std::string s3GetHostname(
+    irods::plugin_property_map& _prop_map ) 
 {
-    if (g_hostname.empty())
+
+
+    std::vector<std::string> hostname_vector;
+    size_t hostname_index = 0;
+    _prop_map.get<std::vector<std::string> >(s3_default_hostname_vector, hostname_vector);
+    _prop_map.get<size_t>(s3_hostname_index, hostname_index);
+    if (hostname_vector.empty()) {
         return NULL; // Short-circuit default case
+    }
+
     g_hostnameIdxLock.lock();
```

Comment:
Is the lock needed? The hostname index is retrieved/set elsewhere without a lock, so either they need the lock as well or it is not needed at all.

Reply:
Good point.  I don't believe I changed/removed the lock in the other places.  My guess is they didn't have it in the init function because that was guaranteed to be executed by only one thread at a time.  (?).

Also that lock should protect the hostname_vector as well.  We read the size of the vector outside of any lock.  Will change.